http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -179,27 +204,75 @@ class AzureServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
         case config: RuntimeConfig.AzureConfig => F.pure(config.persistentDiskId)
         case _ =>
           F.raiseError[DiskId](
-            AzureRuntimeHasInvalidRuntimeConfig(cloudContext, runtime.clusterName, context.traceId)
+            AzureRuntimeHasInvalidRuntimeConfig(cloudContext, runtime.clusterName, ctx.traceId)
           )
       }
 
       (hasPermission, wmsResourceId) <- checkSamPermission(cloudContext, runtime.id, runtimeName, userInfo)
 
-      _ <- context.span.traverse(s => F.delay(s.addAnnotation("Done auth call for delete azure runtime permission")))
+      _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done auth call for delete azure runtime permission")))
       _ <- F
         .raiseError[Unit](RuntimeNotFoundException(cloudContext, runtimeName, "permission denied"))
         .whenA(!hasPermission)
 
-      _ <- clusterQuery.markPendingDeletion(runtime.id, context.now).transaction
+      _ <- clusterQuery.markPendingDeletion(runtime.id, ctx.now).transaction
 
       // For now, azure disk life cycle is tied to vm life cycle and incompatible with disk routes
-      _ <- persistentDiskQuery.markPendingDeletion(diskId, context.now).transaction
+      _ <- persistentDiskQuery.markPendingDeletion(diskId, ctx.now).transaction
 
       _ <- publisherQueue.offer(
-        DeleteAzureRuntimeMessage(runtime.id, Some(diskId), workspaceId, wmsResourceId, Some(context.traceId))
+        DeleteAzureRuntimeMessage(runtime.id, Some(diskId), workspaceId, wmsResourceId, Some(ctx.traceId))
       )
     } yield ()
 
+  override def listRuntimes(
+    userInfo: UserInfo,
+    workspaceId: Option[WorkspaceId],
+    cloudProvider: Option[CloudProvider],
+    params: Map[String, String]
+  )(implicit as: Ask[F, AppContext]): F[Vector[ListRuntimeResponse2]] =
+    for {
+      paramMap <- F.fromEither(processListParameters(params))
+
+      runtimes <- RuntimeServiceDbQueries.listRuntimes(paramMap._1, paramMap._2, workspaceId, cloudProvider).transaction
+
+      (runtimesUserIsCreator, runtimesUserIsNotCreator) = runtimes.partition(_.auditInfo.creator == userInfo.userEmail)
+
+      // Here, we check if backleo has updated the runtime sam id with the wsm resource's UUID via type conversion
+      // If it has, we can then use the samResourceId of the runtime (which is the same as the wsm resource id) for permission lookup
+      // If not, we don't have the right sam id anyways to look up the vm at this stage
+      runtimeSamIds = runtimesUserIsNotCreator
+        .map { r =>
+          for {
+            uuid <- Either.catchNonFatal(UUID.fromString(r.samResource.resourceId))
+          } yield WsmResourceSamResourceId(WsmControlledResourceId(uuid))
+        }
+        .flatMap(either =>
+          either match {
+            case Right(id) => List(id)
+            case Left(_)   => List.empty
+          }
+        )
```

Comment:
this means /api/v2/runtimes won't show runtimes whose sam resource Id is not UUID. Can we make sure that in prod, all active runtimes do have UUID as Sam resource ID?

Reply:
Hmm... that doesn't seem to be the case.

I'm actually realizing this logic is pretty bad as it assumes all runtimes in the future will be registered with WSM, which isn't the case. Unfortunately, I think we will need 2 sam calls for this to:
1.  support azure runtimes registered as a WSM controlled resource
2. lookup runtimes Leo has registered in sam (which we have no plans to migrate for google runtimes moving forward) 

I'll push up the code shortly